### DISCUSSION TITLE
SB5.1 - OCI - Deploy/Kubernetes/KeycloakRealm failing

### DISCUSSION THREAD

**Discussion by ddevadat** [Sunbird-Ed/Community#322] [Installation] [answered]
Job: Deploy/Kubernetes/KeycloakRealm
tag: release-5.0.0_RC2

When executed the first time ..the job failed with  the error

`
TASK [keycloak-deploy : Initialize python library to run keycloak bootstrap script] ***
fatal: [10.0.1.6]: FAILED! => {
    "changed": true,
    "cmd": "cd /tmp/python-keycloak-0.12.0 && python setup.py install",
    "delta": "0:00:06.990780",
    "end": "2022-12-28 10:49:55.963173",
    "rc": 1,
    "start": "2022-12-28 10:49:48.972393"
}

STDOUT:

running install
running bdist_egg
running egg_info
writing requirements to python_keycloak.egg-info/requires.txt
writing python_keycloak.egg-info/PKG-INFO
writing top-level names to python_keycloak.egg-info/top_level.txt
writing dependency_links to python_keycloak.egg-info/dependency_links.txt
reading manifest file 'python_keycloak.egg-info/SOURCES.txt'
writing manifest file 'python_keycloak.egg-info/SOURCES.txt'
installing library code to build/bdist.linux-x86_64/egg
running install_lib
running build_py
creating build
creating build/lib.linux-x86_64-2.7
creating build/lib.linux-x86_64-2.7/keycloak
copying keycloak/keycloak_patch.py -> build/lib.linux-x86_64-2.7/keycloak
copying keycloak/__init__.py -> build/lib.linux-x86_64-2.7/keycloak
copying keycloak/urls_patterns.py -> build/lib.linux-x86_64-2.7/keycloak
copying keycloak/connection.py -> build/lib.linux-x86_64-2.7/keycloak
copying keycloak/keycloak_openid.py -> build/lib.linux-x86_64-2.7/keycloak
copying keycloak/exceptions.py -> build/lib.linux-x86_64-2.7/keycloak
copying keycloak/keycloak_admin.py -> build/lib.linux-x86_64-2.7/keycloak
copying keycloak/keycloak_main.py -> build/lib.linux-x86_64-2.7/keycloak
creating build/lib.linux-x86_64-2.7/keycloak/authorization
copying keycloak/authorization/role.py -> build/lib.linux-x86_64-2.7/keycloak/authorization
copying keycloak/authorization/__init__.py -> build/lib.linux-x86_64-2.7/keycloak/authorization
copying keycloak/authorization/permission.py -> build/lib.linux-x86_64-2.7/keycloak/authorization
copying keycloak/authorization/policy.py -> build/lib.linux-x86_64-2.7/keycloak/authorization
creating build/lib.linux-x86_64-2.7/keycloak/tests
copying keycloak/tests/__init__.py -> build/lib.linux-x86_64-2.7/keycloak/tests
copying keycloak/tests/test_connection.py -> build/lib.linux-x86_64-2.7/keycloak/tests
creating build/bdist.linux-x86_64
creating build/bdist.linux-x86_64/egg
creating build/bdist.linux-x86_64/egg/keycloak
copying build/lib.linux-x86_64-2.7/keycloak/keycloak_patch.py -> build/bdist.linux-x86_64/egg/keycloak
creating build/bdist.linux-x86_64/egg/keycloak/tests
copying build/lib.linux-x86_64-2.7/keycloak/tests/__init__.py -> build/bdist.linux-x86_64/egg/keycloak/tests
copying build/lib.linux-x86_64-2.7/keycloak/tests/test_connection.py -> build/bdist.linux-x86_64/egg/keycloak/tests
copying build/lib.linux-x86_64-2.7/keycloak/__init__.py -> build/bdist.linux-x86_64/egg/keycloak
copying build/lib.linux-x86_64-2.7/keycloak/urls_patterns.py -> build/bdist.linux-x86_64/egg/keycloak
copying build/lib.linux-x86_64-2.7/keycloak/connection.py -> build/bdist.linux-x86_64/egg/keycloak
creating build/bdist.linux-x86_64/egg/keycloak/authorization
copying build/lib.linux-x86_64-2.7/keycloak/authorization/role.py -> build/bdist.linux-x86_64/egg/keycloak/authorization
copying build/lib.linux-x86_64-2.7/keycloak/authorization/__init__.py -> build/bdist.linux-x86_64/egg/keycloak/authorization
copying build/lib.linux-x86_64-2.7/keycloak/authorization/permission.py -> build/bdist.linux-x86_64/egg/keycloak/authorization
copying build/lib.linux-x86_64-2.7/keycloak/authorization/policy.py -> build/bdist.linux-x86_64/egg/keycloak/authorization
copying build/lib.linux-x86_64-2.7/keycloak/keycloak_openid.py -> build/bdist.linux-x86_64/egg/keycloak
copying build/lib.linux-x86_64-2.7/keycloak/exceptions.py -> build/bdist.linux-x86_64/egg/keycloak
copying build/lib.linux-x86_64-2.7/keycloak/keycloak_admin.py -> build/bdist.linux-x86_64/egg/keycloak
copying build/lib.linux-x86_64-2.7/keycloak/keycloak_main.py -> build/bdist.linux-x86_64/egg/keycloak
byte-compiling build/bdist.linux-x86_64/egg/keycloak/keycloak_patch.py to keycloak_patch.pyc
byte-compiling build/bdist.linux-x86_64/egg/keycloak/tests/__init__.py to __init__.pyc
byte-compiling build/bdist.linux-x86_64/egg/keycloak/tests/test_connection.py to test_connection.pyc
byte-compiling build/bdist.linux-x86_64/egg/keycloak/__init__.py to __init__.pyc
byte-compiling build/bdist.linux-x86_64/egg/keycloak/urls_patterns.py to urls_patterns.pyc
byte-compiling build/bdist.linux-x86_64/egg/keycloak/connection.py to connection.pyc
byte-compiling build/bdist.linux-x86_64/egg/keycloak/authorization/role.py to role.pyc
byte-compiling build/bdist.linux-x86_64/egg/keycloak/authorization/__init__.py to __init__.pyc
byte-compiling build/bdist.linux-x86_64/egg/keycloak/authorization/permission.py to permission.pyc
byte-compiling build/bdist.linux-x86_64/egg/keycloak/authorization/policy.py to policy.pyc
byte-compiling build/bdist.linux-x86_64/egg/keycloak/keycloak_openid.py to keycloak_openid.pyc
byte-compiling build/bdist.linux-x86_64/egg/keycloak/exceptions.py to exceptions.pyc
byte-compiling build/bdist.linux-x86_64/egg/keycloak/keycloak_admin.py to keycloak_admin.pyc
byte-compiling build/bdist.linux-x86_64/egg/keycloak/keycloak_main.py to keycloak_main.pyc
creating build/bdist.linux-x86_64/egg/EGG-INFO
copying python_keycloak.egg-info/PKG-INFO -> build/bdist.linux-x86_64/egg/EGG-INFO
copying python_keycloak.egg-info/SOURCES.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying python_keycloak.egg-info/dependency_links.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying python_keycloak.egg-info/requires.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying python_keycloak.egg-info/top_level.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
creating dist
creating 'dist/python_keycloak-0.12.0-py2.7.egg' and adding 'build/bdist.linux-x86_64/egg' to it
removing 'build/bdist.linux-x86_64/egg' (and everything under it)
Processing python_keycloak-0.12.0-py2.7.egg
Copying python_keycloak-0.12.0-py2.7.egg to /usr/local/lib/python2.7/dist-packages
Adding python-keycloak 0.12.0 to easy-install.pth file

Installed /usr/local/lib/python2.7/dist-packages/python_keycloak-0.12.0-py2.7.egg
Processing dependencies for python-keycloak==0.12.0
Searching for requests==2.20.0
Reading [https://pypi.python.org/simple/requests/](https://pypi.python.org/simple/requests/%1B[0m)
Downloading [https://files.pythonhosted.org/packages/f1/ca/10332a30cb25b627192b4ea272c351bce3ca1091e541245cccbace6051d8/requests-2.20.0-py2.py3-none-any.whl#sha256=a84b8c9ab6239b578f22d1c21d51b696dcfe004032bb80ea832398d6909d7279](https://files.pythonhosted.org/packages/f1/ca/10332a30cb25b627192b4ea272c351bce3ca1091e541245cccbace6051d8/requests-2.20.0-py2.py3-none-any.whl#sha256=a84b8c9ab6239b578f22d1c21d51b696dcfe004032bb80ea832398d6909d7279%1B[0m)
Best match: requests 2.20.0
Processing requests-2.20.0-py2.py3-none-any.whl
Installing requests-2.20.0-py2.py3-none-any.whl to /usr/local/lib/python2.7/dist-packages
writing requirements to /usr/local/lib/python2.7/dist-packages/requests-2.20.0-py2.7.egg/EGG-INFO/requires.txt
Adding requests 2.20.0 to easy-install.pth file

Installed /usr/local/lib/python2.7/dist-packages/requests-2.20.0-py2.7.egg
Searching for python-jose==1.4.0
Reading [https://pypi.python.org/simple/python-jose/](https://pypi.python.org/simple/python-jose/%1B[0m)
Downloading [https://files.pythonhosted.org/packages/8b/df/6744aa5b43ddcdb2d60869f484ab37a72087eaf260d6607f84ed4bbfe23c/python_jose-1.4.0-py2.py3-none-any.whl#sha256=fed56224664af0ebc3947853f1bed23b5609f90c7b02e3dce5ef5757d0301664](https://files.pythonhosted.org/packages/8b/df/6744aa5b43ddcdb2d60869f484ab37a72087eaf260d6607f84ed4bbfe23c/python_jose-1.4.0-py2.py3-none-any.whl#sha256=fed56224664af0ebc3947853f1bed23b5609f90c7b02e3dce5ef5757d0301664%1B[0m)
Best match: python-jose 1.4.0
Processing python_jose-1.4.0-py2.py3-none-any.whl
Installing python_jose-1.4.0-py2.py3-none-any.whl to /usr/local/lib/python2.7/dist-packages
writing requirements to /usr/local/lib/python2.7/dist-packages/python_jose-1.4.0-py2.7.egg/EGG-INFO/requires.txt
Adding python-jose 1.4.0 to easy-install.pth file

Installed /usr/local/lib/python2.7/dist-packages/python_jose-1.4.0-py2.7.egg
Searching for httmock==1.2.5
Reading [https://pypi.python.org/simple/httmock/](https://pypi.python.org/simple/httmock/%1B[0m)
Downloading [https://files.pythonhosted.org/packages/95/87/5ab157945eb0c552896ed131a01f542a0dec42e4651e7037d97bc52043de/httmock-1.2.5.tar.gz#sha256=70af27cd85b68005dc45ae4b639d7043d16edcde58ae33597d9f714d4e8210eb](https://files.pythonhosted.org/packages/95/87/5ab157945eb0c552896ed131a01f542a0dec42e4651e7037d97bc52043de/httmock-1.2.5.tar.gz#sha256=70af27cd85b68005dc45ae4b639d7043d16edcde58ae33597d9f714d4e8210eb%1B[0m)
Best match: httmock 1.2.5
Processing httmock-1.2.5.tar.gz
Writing /tmp/easy_install-NNdWJX/httmock-1.2.5/setup.cfg
Running httmock-1.2.5/setup.py -q bdist_egg --dist-dir /tmp/easy_install-NNdWJX/httmock-1.2.5/egg-dist-tmp-rhKd5p
Moving httmock-1.2.5-py2.7.egg to /usr/local/lib/python2.7/dist-packages
Adding httmock 1.2.5 to easy-install.pth file

Installed /usr/local/lib/python2.7/dist-packages/httmock-1.2.5-py2.7.egg
Searching for urllib3<1.25,>=1.21.1
Reading [https://pypi.python.org/simple/urllib3/](https://pypi.python.org/simple/urllib3/%1B[0m)
Downloading [https://files.pythonhosted.org/packages/01/11/525b02e4acc0c747de8b6ccdab376331597c569c42ea66ab0a1dbd36eca2/urllib3-1.24.3-py2.py3-none-any.whl#sha256=a637e5fae88995b256e3409dc4d52c2e2e0ba32c42a6365fee8bbd2238de3cfb](https://files.pythonhosted.org/packages/01/11/525b02e4acc0c747de8b6ccdab376331597c569c42ea66ab0a1dbd36eca2/urllib3-1.24.3-py2.py3-none-any.whl#sha256=a637e5fae88995b256e3409dc4d52c2e2e0ba32c42a6365fee8bbd2238de3cfb%1B[0m)
Best match: urllib3 1.24.3
Processing urllib3-1.24.3-py2.py3-none-any.whl
Installing urllib3-1.24.3-py2.py3-none-any.whl to /usr/local/lib/python2.7/dist-packages
writing requirements to /usr/local/lib/python2.7/dist-packages/urllib3-1.24.3-py2.7.egg/EGG-INFO/requires.txt
Adding urllib3 1.24.3 to easy-install.pth file

Installed /usr/local/lib/python2.7/dist-packages/urllib3-1.24.3-py2.7.egg
Searching for idna<2.8,>=2.5
Reading [https://pypi.python.org/simple/idna/](https://pypi.python.org/simple/idna/%1B[0m)
Downloading [https://files.pythonhosted.org/packages/4b/2a/0276479a4b3caeb8a8c1af2f8e4355746a97fab05a372e4a2c6a6b876165/idna-2.7-py2.py3-none-any.whl#sha256=156a6814fb5ac1fc6850fb002e0852d56c0c8d2531923a51032d1b70760e186e](https://files.pythonhosted.org/packages/4b/2a/0276479a4b3caeb8a8c1af2f8e4355746a97fab05a372e4a2c6a6b876165/idna-2.7-py2.py3-none-any.whl#sha256=156a6814fb5ac1fc6850fb002e0852d56c0c8d2531923a51032d1b70760e186e%1B[0m)
Best match: idna 2.7
Processing idna-2.7-py2.py3-none-any.whl
Installing idna-2.7-py2.py3-none-any.whl to /usr/local/lib/python2.7/dist-packages
Adding idna 2.7 to easy-install.pth file

Installed /usr/local/lib/python2.7/dist-packages/idna-2.7-py2.7.egg
Searching for chardet<3.1.0,>=3.0.2
Reading [https://pypi.python.org/simple/chardet/](https://pypi.python.org/simple/chardet/%1B[0m)
Downloading [https://files.pythonhosted.org/packages/bc/a9/01ffebfb562e4274b6487b4bb1ddec7ca55ec7510b22e4c51f14098443b8/chardet-3.0.4-py2.py3-none-any.whl#sha256=fc323ffcaeaed0e0a02bf4d117757b98aed530d9ed4531e3e15460124c106691](https://files.pythonhosted.org/packages/bc/a9/01ffebfb562e4274b6487b4bb1ddec7ca55ec7510b22e4c51f14098443b8/chardet-3.0.4-py2.py3-none-any.whl#sha256=fc323ffcaeaed0e0a02bf4d117757b98aed530d9ed4531e3e15460124c106691%1B[0m)
Best match: chardet 3.0.4
Processing chardet-3.0.4-py2.py3-none-any.whl
Installing chardet-3.0.4-py2.py3-none-any.whl to /usr/local/lib/python2.7/dist-packages
Adding chardet 3.0.4 to easy-install.pth file
Installing chardetect script to /usr/local/bin

Installed /usr/local/lib/python2.7/dist-packages/chardet-3.0.4-py2.7.egg
Searching for certifi>=2017.4.17
Reading [https://pypi.python.org/simple/certifi/](https://pypi.python.org/simple/certifi/%1B[0m)
Downloading [https://files.pythonhosted.org/packages/37/f7/2b1b0ec44fdc30a3d31dfebe52226be9ddc40cd6c0f34ffc8923ba423b69/certifi-2022.12.7.tar.gz#sha256=35824b4c3a97115964b408844d64aa14db1cc518f6562e8d7261699d1350a9e3](https://files.pythonhosted.org/packages/37/f7/2b1b0ec44fdc30a3d31dfebe52226be9ddc40cd6c0f34ffc8923ba423b69/certifi-2022.12.7.tar.gz#sha256=35824b4c3a97115964b408844d64aa14db1cc518f6562e8d7261699d1350a9e3%1B[0m)
Best match: certifi 2022.12.7
Processing certifi-2022.12.7.tar.gz
Writing /tmp/easy_install-jlvgLJ/certifi-2022.12.7/setup.cfg
Running certifi-2022.12.7/setup.py -q bdist_egg --dist-dir /tmp/easy_install-jlvgLJ/certifi-2022.12.7/egg-dist-tmp-rZApK4
creating /usr/local/lib/python2.7/dist-packages/certifi-2022.12.7-py2.7.egg
Extracting certifi-2022.12.7-py2.7.egg to /usr/local/lib/python2.7/dist-packages
Adding certifi 2022.12.7 to easy-install.pth file

Installed /usr/local/lib/python2.7/dist-packages/certifi-2022.12.7-py2.7.egg
Searching for six<2.0
Reading [https://pypi.python.org/simple/six/](https://pypi.python.org/simple/six/%1B[0m)
Downloading [https://files.pythonhosted.org/packages/d9/5a/e7c31adbe875f2abbb91bd84cf2dc52d792b5a01506781dbcf25c91daf11/six-1.16.0-py2.py3-none-any.whl#sha256=8abb2f1d86890a2dfb989f9a77cfcfd3e47c2a354b01111771326f8aa26e0254](https://files.pythonhosted.org/packages/d9/5a/e7c31adbe875f2abbb91bd84cf2dc52d792b5a01506781dbcf25c91daf11/six-1.16.0-py2.py3-none-any.whl#sha256=8abb2f1d86890a2dfb989f9a77cfcfd3e47c2a354b01111771326f8aa26e0254%1B[0m)
Best match: six 1.16.0
Processing six-1.16.0-py2.py3-none-any.whl
Installing six-1.16.0-py2.py3-none-any.whl to /usr/local/lib/python2.7/dist-packages
Adding six 1.16.0 to easy-install.pth file

Installed /usr/local/lib/python2.7/dist-packages/six-1.16.0-py2.7.egg
Searching for pycrypto<2.7.0,>=2.6.0
Reading [https://pypi.python.org/simple/pycrypto/](https://pypi.python.org/simple/pycrypto/%1B[0m)
Downloading [https://files.pythonhosted.org/packages/60/db/645aa9af249f059cc3a368b118de33889219e0362141e75d4eaf6f80f163/pycrypto-2.6.1.tar.gz#sha256=f2ce1e989b272cfcb677616763e0a2e7ec659effa67a88aa92b3a65528f60a3c](https://files.pythonhosted.org/packages/60/db/645aa9af249f059cc3a368b118de33889219e0362141e75d4eaf6f80f163/pycrypto-2.6.1.tar.gz#sha256=f2ce1e989b272cfcb677616763e0a2e7ec659effa67a88aa92b3a65528f60a3c%1B[0m)
Best match: pycrypto 2.6.1
Processing pycrypto-2.6.1.tar.gz
Writing /tmp/easy_install-6aaT0Q/pycrypto-2.6.1/setup.cfg
Running pycrypto-2.6.1/setup.py -q bdist_egg --dist-dir /tmp/easy_install-6aaT0Q/pycrypto-2.6.1/egg-dist-tmp-r5XVj3

STDERR:

zip_safe flag not set; analyzing archive contents...
zip_safe flag not set; analyzing archive contents...
/usr/lib/python2.7/dist-packages/setuptools/dist.py:397: UserWarning: Normalizing '2022.12.07' to '2022.12.7'
  normalized_version,
warning: no previously-included files found matching '.github/'
warning: manifest_maker: MANIFEST.in, line 4: 'recursive-exclude' expects <dir> <pattern1> <pattern2> ...

  File "build/bdist.linux-x86_64/egg/certifi/core.py", line 17
    def where() -> str:
                ^
SyntaxError: invalid syntax

  File "/usr/local/lib/python2.7/dist-packages/certifi-2022.12.7-py2.7.egg/certifi/core.py", line 17
    def where() -> str:
                ^
SyntaxError: invalid syntax

src/_fastmath.c:31:10: fatal error: Python.h: No such file or directory
 #include "Python.h"
          ^~~~~~~~~~
compilation terminated.
error: Setup script exited with error: command 'x86_64-linux-gnu-gcc' failed with exit status 1

MSG:

non-zero return code
`

**Comment by ddevadat** [comment]
When try to re run ..it failed at the checks

`
RUNNING HANDLER [keycloak-deploy : wait for keycloak to start] *****************
fatal: [10.0.1.6]: FAILED! => {}

MSG:

The conditional check '_result.status == 200' failed. The error was: error while evaluating conditional (_result.status == 200): 'dict object' has no attribute 'status'


NO MORE HOSTS LEFT *************************************************************

PLAY RECAP *********************************************************************
10.0.1.6                   : ok=3    changed=2    unreachable=0    failed=1    skipped=0    rescued=0    ignored=1   

`

**Comment by ddevadat** [comment]
Followed the below on the dp host

`
cd /usr/local/lib/python2.7/dist-packages
rm -rf certifi-2022.12.7-py2.7.egg
rm -rf urllib3-1.24.3-py2.7.egg
rm -rf requests-2.20.0-py2.7.egg
`

After this it was able to go past the check. But everytime the job failed ..i have to delete the 3 python packages

**Comment by ddevadat** [comment]
For the error message

`
compilation terminated.
error: Setup script exited with error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
`

I tried with
`
sudo apt-get install python-dev 

`

This now is giving a new error

`
TASK [keycloak-deploy : Run the keycloak bootstrap script] *********************
fatal: [10.0.1.6]: FAILED! => {
    "attempts": 1,
    "changed": true,
    "cmd": "cd /tmp/keycloak/ && python keycloak_main.py /tmp/keycloak-bootstrap.conf.json",
    "delta": "0:00:00.102222",
    "end": "2022-12-28 21:25:40.598220",
    "rc": 1,
    "start": "2022-12-28 21:25:40.495998"
}

STDERR:

Traceback (most recent call last):
  File "keycloak_main.py", line 3, in <module>
    from keycloak import KeycloakOpenID
  File "build/bdist.linux-x86_64/egg/keycloak/__init__.py", line 18, in <module>
    from .keycloak_openid import *
  File "build/bdist.linux-x86_64/egg/keycloak/keycloak_openid.py", line 18, in <module>
    from .authorization import Authorization
  File "build/bdist.linux-x86_64/egg/keycloak/authorization/__init__.py", line 22, in <module>
    
  File "build/bdist.linux-x86_64/egg/keycloak/authorization/policy.py", line 18, in <module>
  File "build/bdist.linux-x86_64/egg/keycloak/exceptions.py", line 18, in <module>
    import requests
  File "/usr/local/lib/python2.7/dist-packages/requests-2.20.0-py2.7.egg/requests/__init__.py", line 112, in <module>
    from . import utils
  File "/usr/local/lib/python2.7/dist-packages/requests-2.20.0-py2.7.egg/requests/utils.py", line 24, in <module>
    from . import certs
  File "/usr/local/lib/python2.7/dist-packages/requests-2.20.0-py2.7.egg/requests/certs.py", line 15, in <module>
    from certifi import where
  File "/usr/local/lib/python2.7/dist-packages/certifi-2022.12.7-py2.7.egg/certifi/__init__.py", line 1, in <module>
    from .core import contents, where
  File "/usr/local/lib/python2.7/dist-packages/certifi-2022.12.7-py2.7.egg/certifi/core.py", line 17
    def where() -> str:
                ^
SyntaxError: invalid syntax


MSG:

non-zero return code
 
`

**Comment by ddevadat** [comment]
could you please provide any pointers on how to take this further.. i am suspecting some compatibility issue with python2

On DP node

`
root@dp:/tmp/keycloak# python --version
Python 2.7.17
root@dp:/tmp/keycloak# python3 --version
Python 3.6.9

root@dp:/tmp/keycloak# pip3 --version
pip 9.0.1 from /usr/lib/python3/dist-packages (python 3.6)

root@dp:/tmp/keycloak# pip3 list
DEPRECATION: The default format will switch to columns in the future. You can use --format=(legacy|columns) (or define a format=(legacy|columns) in your pip.conf under the [list] section) to disable this warning.
asn1crypto (0.24.0)
attrs (17.4.0)
Automat (0.6.0)
blinker (1.4)
certifi (2018.1.18)
chardet (3.0.4)
click (6.7)
cloud-init (22.3.4)
colorama (0.3.7)
command-not-found (0.3)
configobj (5.0.6)
constantly (15.1.0)
cryptography (2.1.4)
distro-info (0.18ubuntu0.18.04.1)
httplib2 (0.9.2)
hyperlink (17.3.1)
idna (2.6)
incremental (16.10.1)
Jinja2 (2.10)
jsonpatch (1.16)
jsonpointer (1.10)
jsonschema (2.6.0)
keyring (10.6.0)
keyrings.alt (3.0)
language-selector (0.1)
MarkupSafe (1.0)
netifaces (0.10.4)
oauthlib (2.0.6)
PAM (0.4.2)
pexpect (4.2.1)
pip (9.0.1)
pyasn1 (0.4.2)
pyasn1-modules (0.2.1)
pycrypto (2.6.1)
pygobject (3.26.1)
PyJWT (1.5.3)
pyOpenSSL (17.5.0)
pyserial (3.4)
python-apt (1.6.5+ubuntu0.7)
python-debian (0.1.32)
pyxdg (0.25)
PyYAML (3.12)
requests (2.18.4)
requests-unixsocket (0.1.5)
SecretStorage (2.3.1)
service-identity (16.0.0)
setuptools (39.0.1)
six (1.11.0)
sos (4.4)
ssh-import-id (5.7)
systemd-python (234)
Twisted (17.9.0)
ubuntu-advantage-tools (27.11.2)
ufw (0.36)
unattended-upgrades (0.1)
urllib3 (1.22)
virtualenv (15.1.0)
wheel (0.30.0)
zope.interface (4.3.2)
`

**Reply by beepdot** [reply]
Can you provide the list of packages installed by pip2?
```
pip2 freeze
```

**Comment by ddevadat** [comment, marked as answer]
The certifi version 2022.12.7 is the one which cause the problem.

This library, in theory should be good with Python3 but it won't work with Python2

we, as a workaround, should manually remove it and install an old version

`
ubuntu@dp:~$ sudo pip uninstall certifi
The directory '/home/ubuntu/.cache/pip/http' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
Uninstalling certifi-2022.12.7:
  /usr/local/lib/python2.7/dist-packages/certifi-2022.12.7-py2.7.egg
Proceed (y/n)? y
  Successfully uninstalled certifi-2022.12.7
ubuntu@dp:~$ sudo pip install certifi==2021.10.8
The directory '/home/ubuntu/.cache/pip/http' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
The directory '/home/ubuntu/.cache/pip' or its parent directory is not owned by the current user and caching wheels has been disabled. check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
Collecting certifi==2021.10.8
  Downloading https://files.pythonhosted.org/packages/37/45/946c02767aabb873146011e665728b680884cd8fe70dde973c640e45b775/certifi-2021.10.8-py2.py3-none-any.whl (149kB)
    100% |¿¿¿¿¿¿¿¿¿¿¿¿¿¿¿¿¿¿¿¿¿¿¿¿¿¿¿¿¿¿¿¿| 153kB 8.3MB/s 
Installing collected packages: certifi
Successfully installed certifi-2021.10.8
// code placeholder
`

After this re executed the job. This time it completed.

**Reply by beepdot** [reply]
Yes, we are aware of this and have also planned a fix. That's why I asked you for `pip2` packages.
@saiakhil46 FYI
